single-line-diagram/single-line-diagram-core/src/test/java/com/powsybl/sld/iidm/TestFeederInfos.java

```diff
@@ -90,6 +91,43 @@ public List<DiagramLabelProvider.NodeDecorator> getNodeDecorators(Node node, Dir
         assertEquals(toString("/TestFeederInfos.svg"), toSVG(g, "/TestFeederInfos.svg", manyFeederInfoProvider, new BasicStyleProvider()));
     }
 
+    @Test
+    public void testAllPossibleInfoItems() {
+        // build graph
+        VoltageLevelGraph g = graphBuilder.buildVoltageLevelGraph(vl.getId());
+
+        layoutParameters.setSpaceForFeederInfos(100)
+                .setFeederInfosIntraMargin(5)
+                .setPowerValuePrecision(0);
+
+        // Run layout
+        voltageLevelGraphLayout(g);
+
+        // many feeder values provider example for the test :
+        DiagramLabelProvider allFeederInfoProvider = new DefaultDiagramLabelProvider(network, componentLibrary, layoutParameters) {
+
+            @Override
+            public List<FeederInfo> getFeederInfos(FeederNode node) {
+                List<FeederInfo> feederInfos = Arrays.asList(
+                        new DirectionalFeederInfo(ARROW_ACTIVE, 1000, valueFormatter::formatPower, null),
+                        new DirectionalFeederInfo(ARROW_REACTIVE, 70, valueFormatter::formatPower, null),
+                        new DirectionalFeederInfo(ARROW_CURRENT, 100, valueFormatter::formatCurrent, null));
+                boolean feederArrowSymmetry = node.getDirection() == Direction.TOP || layoutParameters.isFeederInfoSymmetry();
+                if (!feederArrowSymmetry) {
+                    Collections.reverse(feederInfos);
+                }
+                return feederInfos;
+            }
+
+            @Override
```

Comment:
why don't you use the DefaultDiagramLabelProvider itself?

Reply:
I updated the test to use the DefaultDiagramLabelProvider.